tdrs-backend/tdpservice/search_indexes/parsers/tanf_parser.py

```diff
@@ -0,0 +1,269 @@
+"""Transforms a TANF datafile into an search_index model."""
+
+import logging
+from ..models import T1  # , T2, T3, T4, T5, T6, T7, ParserLog
+# from django.core.exceptions import ValidationError
+from .util import get_record_type
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+class Field:
+    """Provides a mapping between a field name and its position."""
+
+    def __init__(self, name, length, start, end, type):
+        self.name = name
+        self.length = length
+        self.start = start
+        self.end = end
+        self.type = type
+
+    def create(self, name, length, start, end, type):
+        """Create a new field."""
+        return Field(name, length, start, end, type)
+
+    def __repr__(self):
+        """Return a string representation of the field."""
+        return f"{self.name}({self.start}-{self.end})"
+
+class RowSchema:
+    """Maps the schema for data rows."""
+
+    def __init__(self):  # , section):
+        self.fields = []
+        # self.section = section # intended for future use with multiple section objects
+
+    def add_field(self, name, length, start, end, type):
+        """Add a field to the schema."""
+        self.fields.append(
+            Field(name, length, start, end, type)
+        )
+
+    def add_fields(self, fields: list):
+        """Add multiple fields to the schema."""
+        for field, length, start, end, type in fields:
+            self.add_field(field, length, start, end, type)
+
+    def get_field(self, name):
+        """Get a field from the schema."""
+        return self.fields[name]
+
+    def get_field_names(self):
+        """Get all field names from the schema."""
+        return self.fields.keys()
+
+    def get_all_fields(self):
+        """Get all fields from the schema."""
+        return self.fields
+
+
+def active_t1(line, line_number):
```

Comment:
I am having a hard time visualizing what is a line and what is a row. Either the names or doc strings could be more explanative.

Reply:
The function name could be better as well.

Reply:
Haha, I'm open to renaming basically everything. I'll try to expound a little more in the docstrings.

Reply:
@n0remac I have renamed row->line universally (I think). Let me know if you have issues with the docstrings and/or a recommendation on the function name which is presently `active_t1_parser`. Will resolve/move to qasp by COB.